.ci/static-checks.sh

```diff
@@ -838,6 +839,58 @@ static_check_vendor()
 	dep ensure -no-vendor -dry-run
 }
 
+static_check_xml()
+{
+	local all_xml
+	local files
+
+	all_xml=$(find . -name "*.xml" | grep -v "vendor/" | sort || true)
+
+	if [ "$specific_branch" = "true" ]
+	then
+		info "Checking all XML files in $branch branch"
+		files="$all_xml"
+	else
+		info "Checking local branch for changed XML files only"
+
+		local xml_status
+
+		xml_status=$(get_pr_changed_file_details || true)
+		xml_status=$(echo "$xml_status" | grep "\.xml$" || true)
+
+		files=$(echo "$xml_status" | awk '{print $NF}')
+	fi
+
+	[ -z "$files" ] && info "No XML files to check" && return
+
+	local file
+
+	for file in $files
+	do
+		info "Checking XML file '$file'"
+
+		local contents
+
+		# Most XML documents are specified as XML 1.0 since, with the
+		# advent of XML 1.0 (Fifth Edition), XML 1.1 is "almost
+		# redundant" due to XML 1.0 providing the majority of XML 1.1
+		# features. xmllint doesn't support XML 1.1 seemingly for this
+		# reason, so the only check we can do is to (crudely) force
+		# the document to be an XML 1.0 one since XML 1.1 documents
+		# can mostly be represented as XML 1.0.
+		#
+		# This is only really required since Jenkins creates XML 1.1
+		# documents.
+		contents=$(sed "s/xml version='1.1'/xml version='1.0'/g" "$file")
+
+		local ret
+
+		{ echo "$contents" | chronic xmllint -format -; ret=$?; } || true
```

Comment:
nit: I think a `<<< $contents` is technically maybe more efficient than an `echo | ` :-)

Reply:
I guess this might save a few cycles - branch updated.